### PR TITLE
Demo CoreML Partitioner

### PR DESCRIPTION
**Pull request by YifanShenSZ**
## TLDR
This PR adds ViT as an e2e example of leveraging CoreML partitioner

💁‍♂️ Tips for reviewers: Review commit by commit would be easier
1. [Commit 1](https://github.com/pytorch/executorch/pull/1981/commits/96b9289fd8823de38ca7388301f08b061a6e7b2f) is just some polishing
2. [Commit 2](https://github.com/pytorch/executorch/pull/1981/commits/0deef84134193cf4124f7b60a3338b34addbc121) adds the e2e example

## Details: Motivation and Example Setup
CoreML partitioner is intended for non-resident models to benefit from CoreML as much as possible. Concretely, if a model cannot be fully converted to CoreML, then CoreML partitioner would help to have the convertible part run on CoreML backend. Unfortunately, for the models we have tried, we cannot find a perfect "torch exportable but CoreML unconvertible" example: they are either fully CoreML resident, or torch unexportable.

As a result, here we artificially create an example by utilizing the `skip_ops_for_coreml_delegation`. It is still a realistic example:
1. In many use cases, we may not want the initial (or final) embedding part to run on CoreML, especially if they are small
3. Our intended graph break emerges from this example

## Appendix: Exported Program
Use `CoreMLPartitioner()` (fully CoreML-resident)
```
        def forward(self, arg152_1: "f32[1, 3, 224, 224]"):
            # No stacktrace found for following nodes
            lowered_module_0 = self.lowered_module_0
            executorch_call_delegate = torch.ops.higher_order.executorch_call_delegate(lowered_module_0, arg152_1);  lowered_module_0 = arg152_1 = None
            getitem: "f32[1, 1000]" = executorch_call_delegate[0];  executorch_call_delegate = None
            return (getitem,)
```
Use `CoreMLPartitioner(skip_ops_for_coreml_delegation=["aten.convolution.default"])` (intentionally remove `conv` from CoreML)
```
        def forward(self, arg2_1: "f32[768, 3, 16, 16]", arg3_1: "f32[768]", arg152_1: "f32[1, 3, 224, 224]"):
            # File: /Volumes/data/Software/Mine/executorch_121559468/env/lib/python3.10/site-packages/torchvision/models/vision_transformer.py:277 in _process_input, code: x = self.conv_proj(x)
            aten_convolution_default: "f32[1, 768, 14, 14]" = executorch_exir_dialects_edge__ops_aten_convolution_default(arg152_1, arg2_1, arg3_1, [16, 16], [0, 0], [1, 1], False, [0, 0], 1);  arg152_1 = arg2_1 = arg3_1 = None
            
            # No stacktrace found for following nodes
            lowered_module_0 = self.lowered_module_0
            executorch_call_delegate = torch.ops.higher_order.executorch_call_delegate(lowered_module_0, aten_convolution_default);  lowered_module_0 = aten_convolution_default = None
            getitem: "f32[1, 1000]" = executorch_call_delegate[0];  executorch_call_delegate = None
            return (getitem,)
```

### CODE REVIEW

**Comment by YifanShenSZ** [comment]
@shoumikhin Please take a look at [commit 2](https://github.com/pytorch/executorch/pull/1981/commits/0deef84134193cf4124f7b60a3338b34addbc121) and see if this 
is good enough to demo CoreML partitioner :pray: thx
